src/toolkit/matl_buy_policy.cc

```diff
@@ -199,10 +231,32 @@ void MatlBuyPolicy::AcceptMatlTrades(
   for (it = resps.begin(); it != resps.end(); ++it) {
     rsrc_commods_[it->second] = it->first.request->commodity();
     LGH(INFO3) << "got " << it->second->quantity() << " kg of "
-               << it->first.request->commodity();
+               << it->first.request->commodity()  << std::endl;
     buf_->Push(it->second);
   }
 }
 
+void MatlBuyPolicy::SetNextActiveTime() {
+  next_active_end_ = active_dist_->sample() + manager()->context()->time();
+  return;
+};
+
+void MatlBuyPolicy::SetNextDormantTime() {
+  if (next_dormant_end_ < 0) {}
+  else {
+    next_dormant_end_ = dormant_dist_->sample() + next_active_end_;
+  }
+  return;
+}
+
+void MatlBuyPolicy::SetRequestSize() {
+  random_request_size_ = size_dist_->sample();
+  if (random_request_size_ > 1) {
+    random_request_size_ = random_request_size_ / size_dist_->max();
+  }
+  return;
+}
```

Comment:
With the above update, this now should return a `double` and we can eliminate the member variable `random_request_size_` in favor of a local variable here.